lib/Parse/ParseLiteral.cpp

```diff
@@ -142,6 +142,49 @@ std::string Parser::hexUnquote(llvm::StringRef Quoted) {
   return Result;
 }
 
+std::string Parser::simplifyIntegerLiteral(llvm::StringRef Literal) {
+  std::string LiteralStr = Literal.str();
+  LiteralStr.erase(remove(LiteralStr.begin(), LiteralStr.end(), '_'),
+                   LiteralStr.end());
+
+  // Hex format integer literal
+  if (LiteralStr.size() > 2 && LiteralStr[0] == '0' && LiteralStr[1] == 'x')
+    return LiteralStr;
+  size_t eIndex = 0, dotIndex = 0;
+  int eNum = 0, dotNum = 0;
```

Comment:
Variable names should be nouns (as they represent state). The name should be camel case, and start with an upper case letter (e.g. Leader or Boats).